File: common/src/main/java/dev/obscuria/fragmentum/content/registry/DeferredBlockEntity.java
package dev.obscuria.fragmentum.content.registry;

import net.minecraft.core.Holder;
import net.minecraft.world.level.block.entity.BlockEntity;
import net.minecraft.world.level.block.entity.BlockEntityType;

import java.util.function.Supplier;

public class DeferredBlockEntity<T extends BlockEntity> extends Deferred<BlockEntityType<?>, BlockEntityType<T>> {

    public DeferredBlockEntity(Supplier<Holder<BlockEntityType<?>>> supplier) {
        super(supplier);
    }
}

File: common/src/main/kotlin/dev/obscuria/fragmentum/api/registry/DeferredBlockEntity.kt
package dev.obscuria.fragmentum.api.registry

import net.minecraft.core.Holder
import net.minecraft.world.level.block.entity.BlockEntity
import net.minecraft.world.level.block.entity.BlockEntityType
import java.util.function.Supplier

class DeferredBlockEntity<T : BlockEntity>(supplier: Supplier<Holder<BlockEntityType<*>>>)
    : Deferred<BlockEntityType<*>, BlockEntityType<T>>(supplier)